### ISSUE TITLE
Yum state=latest fails if package name or version is too long

### ISSUE THREAD

**Issue by JamesUoM** [ansible/ansible#20043]
##### ISSUE TYPE

 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = 
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT

RedHat 6

##### SUMMARY

yum fails to update a package if the package name or version is too long

I did some digging and found that in the function yum.py: latest() the calls module.run_command("yum check-update") and then process the results line by line using "split.('\n')" however the output from module.run_command() appears to have been formatted to the width of a standard terminal hence:

```
Loaded plugins: product-id, security, subscription-manager

ConsoleKit.x86_64                  0.4.1-6.el6                rhel-6-server-rpms
ConsoleKit-libs.x86_64             0.4.1-6.el6                rhel-6-server-rpms
ImageMagick.x86_64                 6.7.2.7-5.el6_8            rhel-6-server-rpms
Red_Hat_Enterprise_Linux-Release_Notes-6-en-US.noarch
                                   8-2.el6                    rhel-6-server-rpms
abrt.x86_64                        2.0.8-40.el6               rhel-6-server-rpms
abrt-addon-ccpp.x86_64             2.0.8-40.el6               rhel-6-server-rpms
abrt-addon-kerneloops.x86_64       2.0.8-40.el6               rhel-6-server-rpms
...
```

As you can see the tabbed put caused the version and repo for for Red_Hat_Enterprise_Linux-Release_Notes-6-en-US to be forced onto a newline. The same happens if the version number is too long. 

I think yum check-update is using a default width of 80 if no terminal width is set. eg

```
ansible node  -i inventory/ -s -a "tput cols"

node | SUCCESS | rc=0 >>
80
```

##### STEPS TO REPRODUCE

find a package that needs updating whose name is > 30 characters

```bash
ansible node -i inventory/ -s -m yum -a "state=latest name=Red_Hat_Enterprise_Linux-Release_Notes-6-en-US"
```

##### EXPECTED RESULTS

package is updated

##### ACTUAL RESULTS

```
node | SUCCESS => {
    "changed": false, 
    "msg": "", 
    "rc": 0, 
    "results": ["All packages providing Red_Hat_Enterprise_Linux-Release_Notes-6-en-US are up to date", 
""
    ]
}


```

**Comment by JamesUoM** [comment]
Some lame sanitization code that works around the problem:

```
diff --git a/packaging/os/yum.py b/packaging/os/yum.py
index 3d6fed0..d0bd8c8 100644
--- a/packaging/os/yum.py
+++ b/packaging/os/yum.py
@@ -776,7 +776,26 @@ def latest(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
         res['results'].append('Nothing to do here, all packages are up to date')
         return res
     elif rc == 100:
-        available_updates = out.split('\n')
+        raw_available_updates = out.split('\n')
+        # sanitize formatting first as update may be split across lines
+        l = 0
+        prev_line = None
+        available_updates = []
+        for line in raw_available_updates:
+           # skip header
+            if l<2:
+                l += 1
+                continue
+            
+            line_arr = line.split()
+            if len(line_arr) == 3:
+               available_updates.append(line)
+            elif not prev_line:
+               prev_line = line
+            else:
+              available_updates.append("%s %s" % (prev_line, line))
+               prev_line = None
+
         # build update dictionary
         for line in available_updates:
             line = line.split()
```

**Comment by alikins** [comment]
And fixed in https://github.com/ansible/ansible-modules-core/pull/5160 and 
https://github.com/ansible/ansible/pull/21113

**Comment by alikins** [comment]
And additional fixes in https://github.com/ansible/ansible/issues/24330  (merged into devel and backported to stable-2.3)
